src/main/kotlin/graphql/kickstart/tools/SchemaClassScanner.kt

```diff
@@ -333,7 +333,7 @@ internal class SchemaClassScanner(
             }
 
             is InputObjectTypeDefinition -> {
-                graphQLType.inputValueDefinitions.forEach { inputValueDefinition ->
+                (listOf(graphQLType) + inputExtensionDefinitions.filter { it.name == graphQLType.name }).flatMap { it.inputValueDefinitions }.forEach { inputValueDefinition ->
```

Comment:
This is a bit confusing to read. Lets extract the parenthesis into a `inputObjectTypes` variable.
Also please drop a line before each stream method.